Incorrect message about invalid base for a component reference

% f18 -fdebug-semantics cl-test.f90
cl-test.f90:16:3: error: 'g' is an invalid base for a component reference
    g%x=  0.0_8
    ^

MODULE dm
  TYPE gt
    REAL(KIND=8)       :: x
  END TYPE gt
  TYPE(gt)        :: g
END

MODULE cm
  USE dm
  IMPLICIT NONE
CONTAINS
END

SUBROUTINE r()
  USE cm
  g%x=  0.0_8
END

--- a/lib/semantics/resolve-names.cc
+++ b/lib/semantics/resolve-names.cc
@@ -4675,7 +4675,7 @@ const parser::Name *DeclarationVisitor::FindComponent(
   if (!base || !base->symbol) {
     return nullptr;
   }
-  auto &symbol{*base->symbol};
+  auto &symbol{base->symbol->GetUltimate()};
   if (!symbol.has<AssocEntityDetails>() && !ConvertToObjectEntity(symbol)) {
     SayWithDecl(*base, symbol,
         "'%s' is an invalid base for a component reference"_err_en_US);

